packages/opentelemetry-instrumentation-fetch/src/fetch.ts

```diff
@@ -112,7 +112,7 @@ export class FetchInstrumentation extends InstrumentationBase<
   ): void {
     const parsedUrl = web.parseUrl(response.url);
     span.setAttribute(HttpAttribute.HTTP_STATUS_CODE, response.status);
-    if (response.statusText) {
+    if (response.statusText !== undefined) {
```

Comment:
```suggestion
    if (response.statusText != undefined) {
```
Otherwise `null` could slip through.

Reply:
Or `typeof response.statusText === "string"`

Reply:
Same is applicable to all other places where such a check was added.

Reply:
`!=` vs `typeof response.statusText === "string"` will always vote for strict type checking and in this case would rather validate it only for valid cases so `typeof response.statusText === "string"` is definitely a winner for me